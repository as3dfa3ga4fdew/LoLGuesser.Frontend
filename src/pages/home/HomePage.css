.home-container{
  width: 100%;
  height: 100vh;
  padding-top: 80px;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  background: var(--container-color);
}

.home-window{
  width: 600px;
  height: 500px;
  margin-bottom: 100px;
  border: solid 3px var(--window-border-color);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background: var(--window-color);
}

.home-items{
  display: flex;
  flex: row;
}

.home-item{
  border: solid rgb(200, 155, 60) 5px;
  width: 130px;
  height: 130px;
  margin-left: 20px;
  margin-right: 20px;
  display: flex;
  overflow: hidden;
  border-radius: 50%;
  background-color: rgb(200, 170, 110);
  display: flex;
  align-items: center;
  justify-content: center;
  box-shadow: 0 0 20px rgb(200, 155, 60);
}

.lore-image{
  height: 110px;
  width: 110px;
}

.home-titles{
  display: flex;
  flex: row;
  color: white;
  margin-bottom: 10px;
}

.home-title{
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: center;
}