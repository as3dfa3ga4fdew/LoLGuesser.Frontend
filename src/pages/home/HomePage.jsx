import { NavLink } from "react-router-dom";
import "./HomePage.css";
import abilityImage from '../../images/ability.png';
import loreImage from '../../images/lore.png';
import splashImage from '../../images/splash.png';

const HomePage = () =>{
    return(
        <div className="home-container">
            <div className="home-window">
                <div className="home-title">Choose blyad</div>
                <div className="home-items">
                    <NavLink to='/ability' className="home-item">
                        <img src={abilityImage} alt="ability" />
                    </NavLink>
                    <NavLink to='/lore' className="home-item">
                        <img src={loreImage} alt="lore" />
                    </NavLink>
                    <NavLink to='/splash' className="home-item">
                        <img src={splashImage} alt="splash" />
                    </NavLink>
                </div>
            </div>
        </div>
    );
}

export default HomePage;