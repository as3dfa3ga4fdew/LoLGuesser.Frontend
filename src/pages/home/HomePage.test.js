import { React } from 'react';
import HomePage from './HomePage';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';

it('HomePage renders without crashing', () => {
    render(
        <MemoryRouter>
            <HomePage />
        </MemoryRouter>
    );
});

it('Matches previous snapshot', () => {
    const {asFragment} = render(
        <MemoryRouter>
            <HomePage />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it('contains all nav-links', () => {
    const {getAllByRole} = render(
        <MemoryRouter>
            <HomePage />
        </MemoryRouter>
    );
    expect(getAllByRole("link")).toHaveLength(3);
});

it('has the correct CSS classes', () => {
    const {container} = render(
        <MemoryRouter>
            <HomePage />
        </MemoryRouter>
    );
    expect(container.querySelector(".home-container")).toBeInTheDocument();
    expect(container.querySelector(".home-window")).toBeInTheDocument();
});

it("navigates to 'ability' page", () => {
    render(
      <BrowserRouter initialEntries={["/"]}>
        <HomePage />
      </BrowserRouter>
    );
  
    const abilityLink = screen.getByAltText("ability");

    fireEvent.click(abilityLink);
    
    expect(window.location.pathname).toBe("/ability");
  });

it("navigates to 'lore' page", () => {
    render(
        <BrowserRouter>
            <HomePage />
        </BrowserRouter>
    )

    const loreLink = screen.getByAltText("lore");

    fireEvent.click(loreLink);
    
    expect(window.location.pathname).toBe("/lore");
});

it("navigates to 'splash' page", () => {
    render(
        <BrowserRouter>
            <HomePage />
        </BrowserRouter>
    )

    const splashLink = screen.getByAltText("splash");

    fireEvent.click(splashLink);
    
    expect(window.location.pathname).toBe("/splash");
});