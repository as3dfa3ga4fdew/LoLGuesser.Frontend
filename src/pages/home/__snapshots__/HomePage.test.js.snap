// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-container"
  >
    <div
      class="home-window"
    >
      <div
        class="home-items-wrapper"
      >
        <div
          class="home-titles"
        >
          <div
            class="home-title"
          >
            Ability
          </div>
          <div
            class="home-title"
          >
            Lore
          </div>
          <div
            class="home-title"
          >
            Splash Art
          </div>
        </div>
        <div
          class="home-items"
        >
          <a
            class="home-item"
            href="/ability"
          >
            <img
              alt="ability"
              class="lore-image"
              src="ability.png"
            />
          </a>
          <a
            class="home-item"
            href="/lore"
          >
            <img
              alt="lore"
              class="lore-image"
              src="lore.png"
            />
          </a>
          <a
            class="home-item"
            href="/splash"
          >
            <img
              alt="splash"
              class="lore-image"
              src="splash.png"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
